File: app/src/main/java/quintype/com/templatecollectionwithrx/models/story/FacebookUserEngagement.java

package quintype.com.templatecollectionwithrx.models.story;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

public class FacebookUserEngagement {

    @SerializedName("engagement")
    @Expose
    private Integer engagement;
    @SerializedName("shares")
    @Expose
    private Integer shares;
    @SerializedName("comments")
    @Expose
    private Integer comments;
    @SerializedName("reactions")
    @Expose
    private Integer reactions;

    public Integer getEngagement() {
        return engagement;
    }

    public void setEngagement(Integer engagement) {
        this.engagement = engagement;
    }

    public Integer getShares() {
        return shares;
    }

    public void setShares(Integer shares) {
        this.shares = shares;
    }

    public Integer getComments() {
        return comments;
    }

    public void setComments(Integer comments) {
        this.comments = comments;
    }

    public Integer getReactions() {
        return reactions;
    }

    public void setReactions(Integer reactions) {
        this.reactions = reactions;
    }

}

File: app/src/main/java/quintype/com/templatecollectionwithrx/models/story/FacebookUserEngagement.kt
package quintype.com.templatecollectionwithrx.models.story

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName

/**
 * Created TemplateCollectionWithRx by rakshith on 9/4/18.
 */

class FacebookUserEngagement() : Parcelable {

    override fun describeContents(): Int {
        return 0
    }

    @SerializedName("engagement")
    @Expose
    var engagement: Int? = null
    @SerializedName("shares")
    @Expose
    var shares: Int? = null
    @SerializedName("comments")
    @Expose
    var comments: Int? = null
    @SerializedName("reactions")
    @Expose
    var reactions: Int? = null

    constructor(parcel: Parcel) : this() {
        engagement = parcel.readValue(Int::class.java.classLoader) as? Int
        shares = parcel.readValue(Int::class.java.classLoader) as? Int
        comments = parcel.readValue(Int::class.java.classLoader) as? Int
        reactions = parcel.readValue(Int::class.java.classLoader) as? Int
    }

    override fun writeToParcel(parcel: Parcel, flags: Int) {
        parcel.writeValue(engagement)
        parcel.writeValue(shares)
        parcel.writeValue(comments)
        parcel.writeValue(reactions)
    }

    companion object CREATOR : Parcelable.Creator<FacebookUserEngagement> {
        override fun createFromParcel(parcel: Parcel): FacebookUserEngagement {
            return FacebookUserEngagement(parcel)
        }

        override fun newArray(size: Int): Array<FacebookUserEngagement?> {
            return arrayOfNulls(size)
        }
    }
}